tests/shell/mod.rs

```diff
@@ -288,6 +288,22 @@ fn run_in_noninteractive_mode() {
     assert!(child_output.stderr.is_empty());
 }
 
+#[test]
+#[cfg(not(windows))]
+fn run_with_no_newline() {
+    let child_output = std::process::Command::new("sh")
```

Comment:
I'm confused why you're calling `sh` here instead of `nu`, especially since your expect says "failed to run nu".

Reply:
The other tests all do that. I'm going to try to fix them in a separate PR

Reply:
ok, but if we're testing `nu -c blah blah` command line parameters, why are doing `sh -c blah blah`. Is nushell getting invoked in some obscure way to test this? Sorry if I'm missing the point.

Reply:
it's doing `sh -c "nu blah blah"` for some reason. I don't really understand why and that seems to be the reason for the `#[cfg(not(windows))]`. I just copied the structure of the other tests in here, but I don't really believe that it's necessary, so I'm going to try to fix them all

Reply:
ok, I think I understand how this happened - someone probably just copied the `sh -c "nu -h | false"` test, even though that was only necessary to be able to pipe to `false`

can definitely fix it.